### ISSUE TITLE
[Request] Additional player entry field for starting ship name

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#6577]
### Problem Description

In the [Creating-starts](https://github.com/endless-sky/endless-sky/wiki/Creating-Starts) section we have code designated to allow the player to set their name and give:
```
start [<identifier>]
    name <name>
    ship <model>
        {ship specification...}
```

Unfortunately, there doesn't seem to be a way to either set the ship name, or ask the player to name it. In my attempts at doing so in the Lampyrid plugin, the ship is always created with a blank name.

### Related Issue Links

-

### Desired Solution

Two things:
1. The optional ability to specify a name in the start syntax.
```
start "lampyrid"
	ship <ship model here> [<ship name here>]
```

2. A data entry field like that of `name` where the player can type in a name for the ship: `shipname`

### Alternative Approaches

-

### Additional Context

-

**Comment by tehhowch** [comment]
This feels like a duplicate of #5797.
 As [mentioned there] (https://github.com/endless-sky/endless-sky/issues/5797#issuecomment-800567788), you must enter a full ship definition there, not "mission NPC" syntax:

https://github.com/endless-sky/endless-sky/blob/9f91615e8fc581f67d06b7a23c5d0388208ddf7c/source/StartConditions.cpp#L106-L114

So I would expect
```
start
  ship FooModel
    name "Wrath of the Skies"
```
to provide the player a pre-named ship, since the "name" token is parsed by the Ship class to that field:
https://github.com/endless-sky/endless-sky/blob/9f91615e8fc581f67d06b7a23c5d0388208ddf7c/source/Ship.cpp#L199-L200

**Comment by Zitchas** [comment]
Just to clarify, are you saying that is the way they can currently be named? Or is that the way you think would make more sense as the give ship is supposed to be a full ship definition?

**Comment by tehhowch** [comment]
> Just to clarify, are you saying that is the way they can currently be named? Or is that the way you think would make more sense as the give ship is supposed to be a full ship definition?

The manner shown is how one currently may specify the name of the given ship, because the current parsing method only fully supports entire ship definitions, but _happens_ to not break entirely should a pattern akin to a declaration be given instead.

**Comment by Zitchas** [comment]
OK. I'd just like to point out that I can't see any mention of this behavior in the [Creating Ships Page](https://github.com/endless-sky/endless-sky/wiki/CreatingShips) so I was unaware that this functionality existed.

OK, I just tested it, and it works. 

Might I suggest that the "Creating Starts" section be expanded to include this?
```
    ship <model>
        name <This ship's individual name>
        {ship specification...}
```

**Comment by tehhowch** [comment]
> OK. I'd just like to point out that I can't see any mention of this behavior in the [Creating Ships Page](https://github.com/endless-sky/endless-sky/wiki/CreatingShips) so I was unaware that this functionality existed.

As would be expected; it makes no sense for a ship template used by random fleets to bake in a specific name.


> Might I suggest that the "Creating Starts" section be expanded to include this?
> ```
>     ship <model>
>         name <This ship's individual name>
>         {ship specification...}
> ```

The intent of `{ship specification...}` is to convey that all supported nodes may be used there. I'll update the wiki for creating ships to indicate that `name` is a supported token with limited use.

**Comment by tehhowch** [comment]
Can you clean up the issue body to clarify the actionable request? Or if there isn't anything here that is different than the issue I linked earlier, please close it as a duplicate

**Comment by TheGiraffe3** [comment]
Will this be fixed by #12458?

**Comment by Amazinite** [comment]
No. I milestoned this issue for the following request in the "desired solution" section:

> A data entry field like that of name where the player can type in a name for the ship: shipname

**Comment by TheGiraffe3** [comment]
And that would be for adjusting the current flagship's name?

**Comment by Amazinite** [comment]
Correct. A start conversation could ask for the player's name, and for their flagship's name.

Per tehhowch's comment, the rest of this issue is an effective duplicate of #5797, which my PR does directly resolve.
